src/main/java/com/google/sps/servlets/DashboardServlet.java

```diff
@@ -0,0 +1,75 @@
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.sps.servlets.Property;
+import java.lang.reflect.Field; 
+import java.util.HashMap;
+
+@WebServlet("/dashboard")
+public class DashboardServlet extends HttpServlet {
+
+ /**
+  * This route will return a JSON with the percentages of each interaction
+  */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    try {
+      HashMap<String, Double> interactionPercentages = calculatePercentages();
+      
+      String jsonToSend = convertToJson(interactionPercentages);
+
+      response.setContentType("application/json; charset=UTF-8");
+      response.getWriter().println(jsonToSend);
+    
+    } catch (IllegalAccessException exception) {
+      response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+    }
+  }
+
+ /**
+  * This function queries for and calculates the percentages of each interaction.
+  * @return A hashmap with the interactions as keys and the percentages as values.
```

Comment:
don't forget to include parameters & return here

Reply:
hmm, i did include return here and there are no parameters to this method ?

Reply:
I'm having some trouble reading today haha, I think I saw the HashMap and thought it was parameters for some reason whoops